receiver/sqlserverreceiver/scraper_test.go

```diff
@@ -32,45 +32,49 @@ func configureAllScraperMetrics(cfg *Config, enabled bool) {
 	cfg.Metrics.SqlserverBatchRequestRate.Enabled = enabled
 	cfg.Metrics.SqlserverBatchSQLCompilationRate.Enabled = enabled
 	cfg.Metrics.SqlserverBatchSQLRecompilationRate.Enabled = enabled
-
+	cfg.Metrics.SqlserverDatabaseBackupOrRestoreRate.Enabled = enabled
```

Comment:
The changes to this method aren't entirely relevant to this PR, my apologies. I noticed not all metrics were being included in this method so I used it as an opportunity to add all metrics to this list and alphabetize them.

To validate this is the full list, refer to the [generated list of metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/61f8c78d4833dfca588db0fa530df1fc79da8413/receiver/sqlserverreceiver/internal/metadata/generated_metrics.go#L353).